spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestMetadataTables.java

```diff
@@ -171,6 +171,13 @@ public void testPartitionedTable() throws Exception {
     Assert.assertEquals("Metadata table should return one data file", 1, actualDataFiles.size());
     TestHelpers.assertEqualsSafe(filesTableSchema.asStruct(), expectedDataFiles.get(0), actualDataFiles.get(0));
 
+    List<Row> actualPartitionsWithProjection =
```

Comment:
While I like this test I believe we should also add one to https://github.com/apache/iceberg/blob/4ae2002bd46bf8e1c20db03cffc6319237e4d74a/core/src/test/java/org/apache/iceberg/TestMetadataTableScans.java

To make sure we have one in the core module.

Reply:
@singhpk234  ^

Reply:
+1 added a UT for the same, thanks @RussellSpitzer !!!

Reply:
Thanks! Otherwise we wouldn't have an automated test running on this with "Core" only changes.